pkg/apis/aws/v1alpha1/account_types.go

```diff
@@ -290,3 +302,55 @@ func (a *Account) GetCondition(conditionType AccountConditionType) *AccountCondi
 	}
 	return nil
 }
+
+const MaximumAccountConditionTypes = 14
```

Comment:
Yup it's just for the max length. I've added it to ensure that people just don't hand bomb in a load of values into the array without thinking about it - we've abused these strings in the past, re-declaring const strings in many places,  I'm just trying to find a way to reduce that. This may not be the best solution, so if you have other ideas, I'm 100% interested in hearing! :)

Reply:
🤔   I get where you're coming from now.

What happens if we do something like add more values without increasing this number?  Will it break in a way that's apparent that what we're doing is wrong?  I'm more worried that if we do something like this where we are explicitly setting the length (let's use this one as the example of 14) and then we do something like add two more fields and then existing fields just silently drop out of the list (or the new ones never show up in the list) and we don't catch it right away.

Let me run a test on the go playground.

Reply:
Nvm, this looks like it will be a relatively "easy" error to spot and fix: https://go.dev/play/p/nrJ7gQTpwfN

TL;DR: Trying to add 6 elements to a 5-length defined list throws the following build error: `./prog.go:14:3: array index 5 out of bounds [0:5]`

Reply:
The most we could potentially do there is to add a comment to where the values are defined, like here:
https://github.com/openshift/aws-account-operator/pull/603/files#diff-babfe6bb2a6c13e52d68a586e1375736447e90c292c499956cac9e56583a7a37R31

To indicate that the corresponding MAX value should be updated, and the equivalent `getAll*Types()` func needs to have the new value added to its array?

Reply:
Yeah, maybe that would be a good compromise here.  We still get the security of just not adding items to the list, however we also have a comment of what the var does and why we chose to do it that way, for those who come after us.